### DISCUSSION TITLE
List of all variables

### DISCUSSION THREAD

**Discussion by SPancerfaustyna** [AlexProgrammerDE/balena-minecraft-server#169] [Q&A]
Hey, is there any list of all variables? Or can you list all here?

**Comment by viedok** [comment]
The ones I know :

**DEVICE VARIABLES**
`RAM` -> example : "7G" or "2G"
`ENABLE_UPDATE` -> boolean
`SCP_PASSWORD` -> string

**CONFIGURATION VARIABLES**
`BALENA_HOST_CONFIG_arm_freq` -> value
`BALENA_HOST_CONFIG_over_voltage` -> value
_these last ones are for overclocking purposes, if you don't know what you are doing, don't change them!_

**Comment by viedok** [comment]
other ones I just found while reading the start.sh code :

**DEVICE VARIABLES**
`DEVICE_HOSTNAME`
`JAR_FILE`
`ENABLE_CONFIG_UPDATE`
`SERVER_JAR_SHA256`
`MC_VERSION`
`LATEST_BUILD`
`SERVER_JAR_URL`

**Comment by Rigodon8688** [comment]
# Variables

The available built in variables for this project.

## mc-server

| Name                 | Default               | Description                                                                                                                                                            |
|----------------------|-----------------------|------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| RAM                  | 1G                    | The allocated RAM to the Minecraft Server on startup. This can not be as much RAM as the device has. Always keep one gigabyte of RAM or more for the operating system! |
| DEVICE_HOSTNAME      | balenaminecraftserver | Hostname for the device in the local network the server is in                                                                                                          |
| JAR_FILE             | paper.jar             | Name of the file to be executed in the startup command                                                                                                                 |
| FLAGS                |                     | Add custom startup flags to the startup command                                                                                                                        |
| AIKAR_FLAGS          | false                 | Add Aikar startup flags to the startup command                                                                                                                         |
| ZGC_FLAGS            | false                 | Add ZGC startup flags to the startup command                                                                                                                           |
| ENABLE_UPDATE        | false                 | Forces a server jar update                                                                                                                                             |
| ENABLE_CONFIG_UPDATE | false                 | Forces a server config update                                                                                                                                          |

## scp-server

| Name         | Default | Description                                     |
|--------------|---------|-------------------------------------------------|
| SCP_PASSWORD | balena  | SCP password for connecting to the file system. |
